src/devices/MemoryLCD/LSxxxB7DHxx.cs

```
@@ -0,0 +1,256 @@
+            else
+            {
+                _shouldDispose = false;
+                _gpio = null;
+            }
+
+            _scs = chipSelect;
+            _disp = display;
+            _extcomin = externalCom;
+
+            BytesPerLine = (PixelWidth + 7) / 8;
+
+            _lineNumberBuffer = Enumerable.Range(0, PixelHeight).Select(m => (byte)m).ToArray();
+            _frameBuffer = new byte[BytesPerLine * PixelHeight];
+
+            Init();
+        }
+
+        /// <summary>
+        /// Updates data of only one specified line.
+        /// </summary>
+        public void DataUpdate1Line(byte lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {PixelWidth}ck ({BytesPerLine} bytes)");
+            }
+
+            // m(1), ag(1), d(18), dummy(2)
+            Span<byte> outputBuffer = stackalloc byte[BytesPerLine + 4];
+
+            outputBuffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+            outputBuffer[1] = AgBytes.Table[lineIndex];
+            data.CopyTo(outputBuffer.Slice(2, BytesPerLine));
+
+            WriteSpi(outputBuffer);
+        }
+
+        /// <summary>
+        /// Updates arbitrary multiple lines data.
+        /// </summary>
+        public void DataUpdateMultipleLines(Span<byte> lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != lineIndex.Length * BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {lineIndex.Length * PixelWidth}ck ({lineIndex.Length * BytesPerLine} bytes)");
+            }
+
+            // m(1), ag1(1), d1(18), dummy2(1), ag2(1), d2(18), ... , dummyn(1), agn(1), dn(18), dummy(2)
+            Span<byte> outputBuffer = stackalloc byte[2 + (2 + BytesPerLine) * lineIndex.Length];
```

It is dangerous to blindly do stackalloc from size that is provided as a parameter as you could easily hit the max size which would crash the app. Before doing this we should some validation that the data passed in will fit the LCD and fail early if not.